TriggerService/docker_context/TriggerHandler.py

```diff
@@ -44,39 +46,92 @@ def toTrigger(self, trigger_data):
             return trigger
 
     async def fetchExistingTriggers(self):
-        endpoint = '/triggers/get/active'
-        results, status = await self.client.getRequest(f'{self.cache_url}{endpoint}')
-        if status != 200 or results is None:
-            logger.error(f'Failed to fetch existing triggers from database. Service shutting down.')
-            sys.exit(1)
-
-        triggers = self.toTriggers(results)
+        triggers = []
+        # TODO waiting on DBM additions
         return triggers
 
     async def setBuyAmount(self, trans_num, command, user_id, stock_symbol, amount):
-        # TODO make POST request to cache service endpoint
-        pass
+        endpoint = '/triggers/buy/set/amount'
+        data = {'transaction_num': trans_num, 'user_id': user_id, 'stock_symbol': stock_symbol, 'amount': amount}
+
+        results, status = await self.client.postRequest(f'{self.dbm_url}{endpoint}', data)
+        if status != 200 or results is None:
+            logger.error(f'Failed during set buy amount call to DBM: {results}')
+            return "setBuyAmount failed on DBM call", 500
+        return "Success", 200
 
     async def setBuyTrigger(self, trans_num, command, user_id, stock_symbol, amount):
-        # TODO make POST request to cache service endpoint
-        # TODO return trigger object for the execution manager
-        pass
+        endpoint = '/triggers/buy/set/price'
+        data = {'transaction_num': trans_num, 'user_id': user_id, 'stock_symbol': stock_symbol, 'price': amount}
+
+        results, status = await self.client.postRequest(f'{self.dbm_url}{endpoint}', data)
+        if status != 200 or results is None:
+            logger.error(f'Failed during set buy trigger call to DBM: {results}')
+            return "setBuyTrigger failed on DBM call", 500
+
+        trigger_data = results.json
+        trigger_data['command'] = command
+        trigger_data['trigger_price'] = trigger_data.pop('price')
+        trigger = self.toTrigger(trigger_data)
+
+        self.TriggerExecutionManager.addTrigger(trigger)

```

Comment:
do we want to have this logic handled in the service logic? This way we don't end up coupling the handler and execution manager. We could return the trigger object from the back, service logic then adds that to the execution and if all is successful returns the result for the api

Reply:
Changed to do this.